frontend/lib/src/components/elements/PlotlyChart/PlotlyChart.tsx

```diff
@@ -73,65 +120,194 @@ function PlotlyFigure({
+            }
+          })
+        }
+      }
     }
 
     return spec
-  }, [
-    element.theme,
-    element.useContainerWidth,
-    figure.spec,
-    height,
-    theme,
-    width,
-  ])
-
-  const [config, setConfig] = useState(JSON.parse(figure.config))
-  const [spec, setSpec] = useState(generateSpec())
-
-  // Update config and spec references iff the theme or props change
-  // Use useLayoutEffect to synchronize rerender by updating state
-  // More information: https://kentcdodds.com/blog/useeffect-vs-uselayouteffect
-  useLayoutEffect(() => {
-    setConfig(JSON.parse(figure.config))
-    setSpec(generateSpec())
-  }, [element, theme, height, width, figure.config, generateSpec])
+  }, [element, figure.spec, theme, widgetMgr])
+
+  const spec = getInitialValue()
+
+  const [initialHeight] = useState(spec.layout.height)
+  const [initialWidth] = useState(spec.layout.width)
+
+  if (isFullScreen(height)) {
+    spec.layout.width = width
+    spec.layout.height = height
+  } else if (element.useContainerWidth) {
+    spec.layout.width = width
+    if (!isFullScreen(height) && height !== initialHeight) {
+      spec.layout.height = initialHeight
+    }
+  } else {
+    spec.layout.width = initialWidth
+    spec.layout.height = initialHeight
+  }
+  if (element.isSelectEnabled) {
+    spec.layout.clickmode = "event+select"
+    spec.layout.hovermode = "closest"
+  }
+  if (element.theme === "streamlit") {
+    applyStreamlitTheme(spec, theme)
+  } else {
+    // Apply minor theming improvements to work better with Streamlit
+    spec.layout = layoutWithThemeDefaults(spec.layout, theme)
+  }
+
+  const handleSelect = (event: Readonly<Plotly.PlotSelectionEvent>): void => {
+    const returnValue: any = { select: {} }
+    const { data } = spec
+    const pointIndices: number[] = []
+    const selectedBoxes: Selection[] = []
+    const selectedLassos: Selection[] = []
+    const selectedPoints: Array<any> = []
+
+    if (
+      event.points.length === 0 &&
+      // event.selections doesn't show up in the PlotSelectionEvent
+      // @ts-expect-error
+      event.selections &&
+      // event.selections doesn't show up in the PlotSelectionEvent
+      // @ts-expect-error
+      event.selections.length === 0
+    ) {
+      return
+    }
+
+    event.points.forEach(function (point: any) {
+      selectedPoints.push({
+        ...point,
+        legendgroup: point.data.legendgroup
+          ? point.data.legendgroup
+          : undefined,
+        data: undefined,
+        fullData: undefined,
+      })
+      pointIndices.push(point.pointIndex)
+
+      // build graph representation to retain state
+      if (
+        data[point.curveNumber] &&
+        !data[point.curveNumber].selectedpoints.includes(point.pointIndex)
+      ) {
+        data[point.curveNumber].selectedpoints.push(point.pointIndex)
+      } else {
+        data[point.curveNumber].selectedpoints = [point.pointIndex]
+      }
+    })
```

Comment:
I'm not sure if this code is fully correct. Is it expected that: `data[point.curveNumber].selectedpoints = [point.pointIndex]` will get executed if either `data[point.curveNumber]` is falsy or if `point.pointIndex` is not included in `data[point.curveNumber].selectedpoints`?

Reply:
Hm what I want to do here is add each `pointIndex` to the corresponding trace (`data[point.curveNumber]`). 

I attempt to add it if an array already exists. 
If it doesn't initialize the array with the pointIndex. 

Maybe I should check that `data[point.curveNumber]` is not null and undefined rather than falsy?